microverseinc/ror-social-scaffold
# #Milestone-5-Feature/friendshipv1

**What does this PR do?**
This PR is for TSE review  of our social media project milestone-5 implementation.

**Tasks Worked on**
-  send friend request from user's profile page
- send friend request from All users index page
- Accept or reject friend request
-Unit/integration tests for the models and UI

**How can this this  be manually Tested?**
- After cloning this repo, 
- run `bundle install`
- start the server `rails s`
- Head over to the browser on  `localhost:3000`
- sign up at least 2 users, note their email and password
- sign in with the first user => click on `my profile` on the navbar  => select a user from the select list displayed => hit the `send friend request` button => `sign out`
- Sign in with the user you just sent a friend request to
- Then you'll see the `Accept friend request` and `Reject` buttons => click on any of the buttons
- if the `Accept` button was selected, then you cant send friend request to that user again but if the initial friend request was rejected, then you can be able to send friend request again to that user.

- To send a request from the All_users page, click on the `all users` link on the navbar and follow the links to test out the rest of the app
- To run Unit/integration tests, head over to the terminal and run `rspec`
